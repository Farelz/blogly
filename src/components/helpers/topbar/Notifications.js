import React from 'react';
import types from 'prop-types';
import { format_date } from '../../../js/helpers';
import img from '../../../img/default-pic.png';

export const Notifications = ({ notifications }) => (
    <div id="notifications" className="dropdown-container">
        <ul className="dropdown-content" id="notifications-menu">
            {notifications.length > 0 ?
                notifications.map((notification) => (
                    <li 
                        className={'notification ' + (notification.read == false ? 'unread' : '')}
                        key={notification.id}
                    >
                        <div className="row">
                            <div className="col s3">
                                <div className="author-image">
                                    <img src={img} className="responsive-img circle" />
                                </div>
                            </div>
                            <div className="notification-info col s9">
                                <p className="user-action">
                                    {notification.from.name} {' '}
                                    <span>{
                                        notification.type == 'comment' ? 
                                            'commented on': 'clapped for'}
                                    </span> {' '}
                                    {notification.post.title}
                                </p>
                                <p className="post-title">{notification.post.title}</p>
                                <p className="meta">{format_date(notification.timestamp)}</p>
                            </div>
                        </div>
                    </li>
                )) : (
                    <li className="notification">
                        <a className="center">No new notifications</a>
                    </li>
                )}
        </ul>
    </div>
);

Notifications.propTypes = {
    notifications: types.array.isRequired
};
