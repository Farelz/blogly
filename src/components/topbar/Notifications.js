import React from 'react';
import types from 'prop-types';
import Link from 'next/link';
import { format_date, get_page_link } from '../../lib/helpers';

const Notifications = ({ notifications }) => {
    const RenderNotifications = ({ notification }) => {
        const {
            node: { read, post, id, fromAuthor, type, timestamp }
        } = notification;

        const page_link = get_page_link(post.title, post.id);

        return (
            <li className={'notification ' + (read === false ? 'unread' : '')} key={id}>
                <div className="row">
                    <div className="col s3">
                        <div className="author-image">
                            <img
                                alt={fromAuthor.fullName}
                                src={fromAuthor.gravatarUrl || '/static/default-pic.png'}
                                className="responsive-img circle"
                            />
                        </div>
                    </div>
                    <div className="notification-info col s9">
                        <p className="user-action">
                            {fromAuthor.fullName}{' '}
                            <span>{type == 'comment' ? 'commented on' : 'clapped for'}</span>{' '}
                        </p>
                        <p className="post-title">
                            <Link href={`/p/${page_link}`}>
                                <a>{post.title}</a>
                            </Link>
                        </p>
                        <p className="meta">{format_date(timestamp)}</p>
                    </div>
                </div>
            </li>
        );
    };
    RenderNotifications.propTypes = { notification: types.object };

    const RenderNoNotifications = () => {
        return (
            <li className="notification">
                <a className="center">You have no notifications</a>
            </li>
        );
    };

    return (
        <div id="notifications" className="dropdown-container">
            <ul className="dropdown-content" id="notifications-menu">
                {notifications.length > 0 ? (
                    notifications.map((notification) => (
                        <RenderNotifications
                            key={notification.node.id}
                            notification={notification}
                        />
                    ))
                ) : (
                    <RenderNoNotifications />
                )}
            </ul>
        </div>
    );
};

Notifications.propTypes = {
    notifications: types.array.isRequired
};

export default Notifications;
