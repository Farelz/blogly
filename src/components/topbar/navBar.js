import React from 'react';
import types from 'prop-types';
import Link from 'next/link';
import { MdMenu } from 'react-icons/lib/md';

import NotificationsMenu from './notificationMenu';
import UserMenu from './userMenu';

export class NavBar extends React.Component {
    render_menu_icon = () => (
        <a data-target="mobile-topbar" className="sidenav-trigger hide-on-med-and-up">
            <MdMenu />
        </a>
    );
    render_site_title = () => (
        <Link href="/">
            <a className="brand-logo center">
                <span className="blog-title">ReactPress</span>
            </a>
        </Link>
    );
    render() {
        const isLoggedIn = this.props.isLoggedIn;

        const user_menu_props = {
            isLoggedIn,
            imageData: this.props.imageData,
            logout: this.props.logout
        };

        const notifications_props = {
            isLoggedIn,
            notifications_data: this.props.notifications_data,
            readNotifications: this.props.readNotifications
        };

        return (
            <nav className="topbar nav-extended">
                <div className="nav-wrapper">
                    {/* Menu icons for small screens */}
                    {this.render_menu_icon()}

                    {/* SiteTitle */}
                    {this.render_site_title()}

                    {/* show login/signup or user-profile depending on login state */}
                    <UserMenu {...user_menu_props} />

                    {/* show notification - menu for small and large screen*/}
                    <NotificationsMenu {...notifications_props} />
                </div>
            </nav>
        );
    }
}

NavBar.propTypes = {
    isLoggedIn: types.bool.isRequired,
    imageData: types.object,
    notifications_data: types.object.isRequired,
    logout: types.func.isRequired,
    readNotifications: types.func.isRequired
};
