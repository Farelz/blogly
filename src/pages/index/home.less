/**
 * ./src/style/home.css
 */

@import '../../style/variables';

/***************** Body Block*******************/
.home-body {
    padding-top: 5rem;
    width: 80%;

    /* Post card */
    .post-card {
        height: 300px;
        max-height: 300px;

        @media screen and (max-width: 600px) {
            height: inherit;
            max-height: inherit;
            display: block;
        }

        .post-card__image {
            height: 100%;
            max-height: 100%;
            width: 40%;
            max-width: 40%;
            background-size: cover;
            background-position: center;
            background-repeat: no-repeat;

            @media screen and (max-width: 600px) {
                height: 150px;
                max-height: 150px;
                width: 100%;
                max-width: 100%;
            }
        }

        /* Post content children */
        .post-card__title {
            font-size: 1.4em;
            font-weight: 600;
            letter-spacing: -0.7px;
            margin-top: -0.3rem;
            display: -webkit-box;
            -webkit-line-clamp: 2;
            -webkit-box-orient: vertical;
            overflow: hidden;

            .post-card__title--link {
                color: @black-scheme-2;

                &:hover {
                    color: @black-scheme-2;
                }
            }
        }
        .post-card__text {
            color: @black-scheme-3;
            font-size: 16px;
            font-weight: 500;
            letter-spacing: -0.5px;
            display: -webkit-box;
            -webkit-line-clamp: 5;
            -webkit-box-orient: vertical;
            overflow: hidden;
        }

        /* Post action */
        .post-card__meta {
            padding: 9px 24px;
            background: hsl(0, 0%, 99%);
            border-top: none;
            border-bottom: 4px solid @color-scheme-2;

            .post-card__meta__author {
                display: flex;

                .post-card__meta__author-image {
                    max-width: 20%;
                    margin-right: 2px;

                    img {
                        width: 40px;
                    }
                }

                .post-card__meta__info {
                    display: flex;
                    flex-direction: column;
                    margin-left: 10px;

                    .post-card__meta__name {
                        font-weight: 700;
                        font-size: 15.11px;
                        text-transform: capitalize;
                        color: @black-scheme-3;
                    }

                    .post-card__meta__date {
                        font-weight: 500;
                        font-size: 13.432px;
                        color: @black-scheme-3;
                    }
                }
            }
        }
    }

    /* pagination */
    .pagination {
        div {
            border: 1px solid @color-scheme-2;
            border-radius: 2px;
            margin-bottom: 12px;
            cursor: pointer;

            &.next {
                float: right;
            }

            &:hover {
                background: @color-scheme-2;

                a {
                    color: #fff !important;
                }
            }

            p {
                color: @color-scheme-2;
                font-weight: 500;
                text-transform: uppercase;
                text-align: center;
            }
        }
    }

    @media screen and (max-width: 600px) {
        padding-top: 3rem;
    }
}
