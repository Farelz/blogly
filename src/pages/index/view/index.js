import React from 'react';
import types from 'prop-types';
import PostCard from './post-card';
import { sort_posts } from '../../../lib/helpers';

export class Body extends React.Component {
    componentDidMount() {
        window.addEventListener('scroll', this.onScroll, false);
    }
    componentWillUnmount() {
        window.removeEventListener('scroll', this.onScroll, false);
    }
    onScroll = () => {
        if (window.innerHeight + window.scrollY >= document.body.offsetHeight) {
            this.props.fetch_more();
        }
    };

    render_post_cards = (posts) => {
        if (posts.length > 0) {
            return posts.map((obj) => (
                <div key={obj.node.id} className="col l6 m10 s12">
                    <PostCard post={obj.node} />
                </div>
            ));
        } else {
            return (
                <div className="col s12">
                    <h5 className="center">They{'\''}re currently no post available.</h5>
                </div>
            );
        }
    };
    render() {
        const posts = sort_posts(this.props.posts);

        return (
            <div className="body section container">
                <div className="row">{this.render_post_cards(posts)}</div>
            </div>
        );
    }
}

Body.propTypes = {
    posts: types.array.isRequired,
    fetch_more: types.func.isRequired
};

export default Body;
