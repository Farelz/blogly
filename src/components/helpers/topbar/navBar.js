import React from 'react';
import types from 'prop-types';
import { Link} from 'react-router-dom';
import { MdMenu, MdNotificationsNone, MdNotificationsActive } from 'react-icons/lib/md';
import { UserDropDownMenu } from './UserDropdown';
import { Notifications } from './Notifications';
import { NavMenu } from './navMenu';

export class NavBar extends React.Component {
    constructor(props){
        super(props);

        this.notificationDropdownTrigger = React.createRef();
        this.userDropDownTrigger = React.createRef();
        this.notificationDropdownIns;
        this.userDropDownIns;
    }
    componentDidMount(){
        this.register_dropdown(
            this.notificationDropdownTrigger.current, 
            'DIV#notifications',
            'notificationDropdownIns',
            {onCloseEnd: () => this.props.readNotifications()}
        );
    }
    componentDidUpdate(){
        this.register_dropdown(
            this.userDropDownTrigger.current,
            'DIV#navbar-user-profile', 
            'userDropDownIns'
        );
    }
    componentWillUnmount() {
        this.deactivate_dropdown('notificationDropdownIns');
        this.deactivate_dropdown('userDropDownIns');
    }
    deactivate_dropdown = (ins) => {
        if (this[ins] !== undefined && this[ins] !== null) {
            this[ins].close();
            this[ins].destroy();
        }
    };
    register_dropdown = (elem, container, ins, extraOptions = {}) => {
        if (elem != null && window.M !== undefined) {
            this[ins] = window.M.Dropdown.init(elem, {
                constrainWidth: false,
                coverTrigger: false,
                container,
                ...extraOptions
            });
        }
    };
    render() {
        const menu_class = this.props.stick ? ' sticky' : '';
        return (
            <nav className="topbar nav-extended">
                <div className="nav-wrapper">
                    {/* Menu icons for small screens */}
                    <a
                        href="#"
                        data-target="mobile-topbar"
                        className="sidenav-trigger hide-on-med-and-up"
                    >
                        <MdMenu />
                    </a>
    
                    {/* SiteTitle */}
                    <Link to="/" className="brand-logo center">
                        <span className="blog-title">ReactPress</span>
                    </Link>

                    {/* show login/signup or user-profile depending on login state */}
                    <ul className="right hide-on-small-only">
                        {this.props.isLoggedIn ? (
                            <li>
                                <a
                                    ref={this.userDropDownTrigger}
                                    className="dropdown-trigger user-profile"
                                    data-target="dropdown-menu"
                                >
                                    <img
                                        className={
                                            'user-image responsive-img circle' + 
                                            this.props.imageData.imageClass}
                                        src={this.props.imageData.image}
                                    />
                                </a>
                                <UserDropDownMenu handleLogout={this.props.logout} />
                            </li>
                        ) : (
                            <React.Fragment>
                                <li>
                                    <Link to="/auth/login" className="sign-in">
                                        Sign in
                                    </Link>
                                </li>
                                <li>
                                    <Link to="/auth/signup" className="sign-up btn">
                                        Get started
                                    </Link>
                                </li>
                            </React.Fragment>
                        )}
                    </ul>
    
                    {/* show notification - menu for small and large screen*/}
                    {/* show only when loggedIn */}
                    {this.props.isLoggedIn && (
                        <ul className="right">
                            <li>
                                <a
                                    className="notification-icon"
                                    ref={this.notificationDropdownTrigger}
                                    data-target="notifications-menu"
                                >
                                    {this.props.notifications_data.unread_count > 0 ? (
                                        <MdNotificationsActive />
                                    ) : (
                                        <MdNotificationsNone />
                                    )}
                                </a>
        
                                <Notifications 
                                    notifications={this.props.notifications_data.notifications}/>
                            </li>
                        </ul>
                    )}
                </div>
    
                {/* Don't show menu on post page */}
                {(this.props.isPostPage) && (
                    <div className={'nav-content' + menu_class}>
                        <NavMenu />
                    </div>
                )}
            </nav>
        );
    }
}

NavBar.propTypes = {
    isLoggedIn: types.bool.isRequired,
    isPostPage: types.bool.isRequired,
    stick: types.bool.isRequired,
    imageData: types.object,
    notifications_data: types.object.isRequired,
    logout: types.func.isRequired,
    readNotifications: types.func.isRequired
};
