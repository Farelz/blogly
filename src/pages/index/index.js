import React from 'react';
import types from 'prop-types';
import MainPage from '../../components/app';
import Body from './view/index';
import './home.less';

export const Index = (props) => {
    return (
        <MainPage
            loggedIn={props.loggedIn}
            pageTitle="Homepage"
            render={() => (
                <Body />
            )}
        />
    );
};

Index.propTypes = {
    loggedIn: types.bool.isRequired,
};

export default Index;
