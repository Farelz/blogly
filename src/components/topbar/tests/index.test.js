import React from 'react';
import { shallow } from 'enzyme';

import { TopBar } from '../index';
import { NavBar } from '../navBar';
import NotificationsMenu from '../notificationMenu';
import Notifications from '../Notifications';

describe('topbar tests', () => {
    const logout = jest.fn();
    const readNotifications = jest.fn();
    const topbarProps = {
        isHomePage: false,
        user_data: {
            user: {}
        },
        notifications: {},
        logout,
        readNotifications
    };

    it('renders', () => {
        const wrapper = shallow(<TopBar {...topbarProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders navbar', () => {
        const props = {
            isLoggedIn: false,
            imageData: {},
            notifications_data: {},
            logout,
            readNotifications
        };

        const wrapper = shallow(<NavBar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders notifications menu', () => {
        const props = {
            isLoggedIn: false,
            notifications_data: {},
            readNotifications
        };

        const wrapper = shallow(<NotificationsMenu {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders notifications', () => {
        const props = {
            notifications: []
        };

        const wrapper = shallow(<Notifications {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
