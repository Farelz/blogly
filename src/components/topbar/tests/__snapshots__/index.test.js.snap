// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`topbar tests renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopBar
    isHomePage={false}
    logout={[MockFunction]}
    notifications={Object {}}
    readNotifications={[MockFunction]}
    user_data={
      Object {
        "user": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar
          imageData={
            Object {
              "image": undefined,
              "imageClass": "",
            }
          }
          isLoggedIn={true}
          logout={[Function]}
          notifications_data={Object {}}
          readNotifications={[Function]}
        />,
        <SideNav
          data={
            Object {
              "image": undefined,
              "imageClass": "",
              "isLoggedIn": true,
              "username": undefined,
            }
          }
          logout={[Function]}
        />,
      ],
      "className": "navbar-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageData": Object {
            "image": undefined,
            "imageClass": "",
          },
          "isLoggedIn": true,
          "logout": [Function],
          "notifications_data": Object {},
          "readNotifications": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "image": undefined,
            "imageClass": "",
            "isLoggedIn": true,
            "username": undefined,
          },
          "logout": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar
            imageData={
              Object {
                "image": undefined,
                "imageClass": "",
              }
            }
            isLoggedIn={true}
            logout={[Function]}
            notifications_data={Object {}}
            readNotifications={[Function]}
          />,
          <SideNav
            data={
              Object {
                "image": undefined,
                "imageClass": "",
                "isLoggedIn": true,
                "username": undefined,
              }
            }
            logout={[Function]}
          />,
        ],
        "className": "navbar-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageData": Object {
              "image": undefined,
              "imageClass": "",
            },
            "isLoggedIn": true,
            "logout": [Function],
            "notifications_data": Object {},
            "readNotifications": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "image": undefined,
              "imageClass": "",
              "isLoggedIn": true,
              "username": undefined,
            },
            "logout": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`topbar tests renders navbar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    imageData={Object {}}
    isLoggedIn={false}
    logout={[MockFunction]}
    notifications_data={Object {}}
    readNotifications={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-wrapper"
      >
        <a
          className="sidenav-trigger hide-on-med-and-up"
          data-target="mobile-topbar"
        >
          <MdMenu />
        </a>
        <Link
          href="/"
        >
          <a
            className="brand-logo center"
          >
            <span
              className="blog-title"
            >
              ReactPress
            </span>
          </a>
        </Link>
        <UserMenu
          imageData={Object {}}
          isLoggedIn={false}
          logout={[MockFunction]}
        />
        <NotificationMenu
          isLoggedIn={false}
          notifications_data={Object {}}
          readNotifications={[MockFunction]}
        />
      </div>,
      "className": "topbar nav-extended",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="sidenav-trigger hide-on-med-and-up"
            data-target="mobile-topbar"
          >
            <MdMenu />
          </a>,
          <Link
            href="/"
          >
            <a
              className="brand-logo center"
            >
              <span
                className="blog-title"
              >
                ReactPress
              </span>
            </a>
          </Link>,
          <UserMenu
            imageData={Object {}}
            isLoggedIn={false}
            logout={[MockFunction]}
          />,
          <NotificationMenu
            isLoggedIn={false}
            notifications_data={Object {}}
            readNotifications={[MockFunction]}
          />,
        ],
        "className": "nav-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MdMenu />,
            "className": "sidenav-trigger hide-on-med-and-up",
            "data-target": "mobile-topbar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a
              className="brand-logo center"
            >
              <span
                className="blog-title"
              >
                ReactPress
              </span>
            </a>,
            "href": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="blog-title"
              >
                ReactPress
              </span>,
              "className": "brand-logo center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ReactPress",
                "className": "blog-title",
              },
              "ref": null,
              "rendered": "ReactPress",
              "type": "span",
            },
            "type": "a",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageData": Object {},
            "isLoggedIn": false,
            "logout": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isLoggedIn": false,
            "notifications_data": Object {},
            "readNotifications": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper"
        >
          <a
            className="sidenav-trigger hide-on-med-and-up"
            data-target="mobile-topbar"
          >
            <MdMenu />
          </a>
          <Link
            href="/"
          >
            <a
              className="brand-logo center"
            >
              <span
                className="blog-title"
              >
                ReactPress
              </span>
            </a>
          </Link>
          <UserMenu
            imageData={Object {}}
            isLoggedIn={false}
            logout={[MockFunction]}
          />
          <NotificationMenu
            isLoggedIn={false}
            notifications_data={Object {}}
            readNotifications={[MockFunction]}
          />
        </div>,
        "className": "topbar nav-extended",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="sidenav-trigger hide-on-med-and-up"
              data-target="mobile-topbar"
            >
              <MdMenu />
            </a>,
            <Link
              href="/"
            >
              <a
                className="brand-logo center"
              >
                <span
                  className="blog-title"
                >
                  ReactPress
                </span>
              </a>
            </Link>,
            <UserMenu
              imageData={Object {}}
              isLoggedIn={false}
              logout={[MockFunction]}
            />,
            <NotificationMenu
              isLoggedIn={false}
              notifications_data={Object {}}
              readNotifications={[MockFunction]}
            />,
          ],
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MdMenu />,
              "className": "sidenav-trigger hide-on-med-and-up",
              "data-target": "mobile-topbar",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="brand-logo center"
              >
                <span
                  className="blog-title"
                >
                  ReactPress
                </span>
              </a>,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="blog-title"
                >
                  ReactPress
                </span>,
                "className": "brand-logo center",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ReactPress",
                  "className": "blog-title",
                },
                "ref": null,
                "rendered": "ReactPress",
                "type": "span",
              },
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "imageData": Object {},
              "isLoggedIn": false,
              "logout": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isLoggedIn": false,
              "notifications_data": Object {},
              "readNotifications": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`topbar tests renders notifications 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notifications
    notifications={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="dropdown-content"
        id="notifications-menu"
      >
        <li
          className="notification"
        >
          <a
            className="center"
          >
            No new notifications
          </a>
        </li>
      </ul>,
      "className": "dropdown-container",
      "id": "notifications",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li
          className="notification"
        >
          <a
            className="center"
          >
            No new notifications
          </a>
        </li>,
        "className": "dropdown-content",
        "id": "notifications-menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="center"
          >
            No new notifications
          </a>,
          "className": "notification",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No new notifications",
            "className": "center",
          },
          "ref": null,
          "rendered": "No new notifications",
          "type": "a",
        },
        "type": "li",
      },
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="dropdown-content"
          id="notifications-menu"
        >
          <li
            className="notification"
          >
            <a
              className="center"
            >
              No new notifications
            </a>
          </li>
        </ul>,
        "className": "dropdown-container",
        "id": "notifications",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li
            className="notification"
          >
            <a
              className="center"
            >
              No new notifications
            </a>
          </li>,
          "className": "dropdown-content",
          "id": "notifications-menu",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="center"
            >
              No new notifications
            </a>,
            "className": "notification",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No new notifications",
              "className": "center",
            },
            "ref": null,
            "rendered": "No new notifications",
            "type": "a",
          },
          "type": "li",
        },
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`topbar tests renders notifications menu 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationMenu
    isLoggedIn={false}
    notifications_data={Object {}}
    readNotifications={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
